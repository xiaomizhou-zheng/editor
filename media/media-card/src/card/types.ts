import React, { MouseEvent, ReactElement, RefObject } from 'react';

import { UIAnalyticsEvent } from '@atlaskit/analytics-next';

import { CardDimensions } from '../types';
import { BreakpointSizeValue } from '../utils/breakpoint';

export interface WrapperProps {
  testId?: string;
  shouldUsePointerCursor?: boolean;
  dimensions?: CardDimensions;
  breakpointSize?: BreakpointSizeValue;
  onClick?: (
    event: React.MouseEvent<HTMLDivElement>,
    analyticsEvent?: UIAnalyticsEvent,
  ) => void;
  onMouseEnter?: (event: MouseEvent<HTMLDivElement>) => void;
  innerRef?: RefObject<HTMLDivElement>;
  children?: JSX.Element;
}

export type InlinePlayerWrapperProps = {
  testId?: string;
  dimensions?: CardDimensions;
  selected: { selected?: boolean | undefined };
  onClick?: (
    event: React.MouseEvent<HTMLDivElement>,
    analyticsEvent?: UIAnalyticsEvent,
  ) => void;
  innerRef?:
    | RefObject<HTMLDivElement>
    | ((instance: HTMLDivElement | null) => void)
    | undefined;
  children?: JSX.Element[] | ReactElement<any, any> | null | any;
};
