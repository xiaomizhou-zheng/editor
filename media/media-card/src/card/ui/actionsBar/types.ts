import { ReactNode } from 'react';
import { CardAction } from '../../actions';

export interface ActionsBarProps {
  actions: Array<CardAction>;
  isFixed?: boolean;
}

export type ActionBarWrapperProps = {
  isFixed?: boolean;
  children?: ReactNode;
};
