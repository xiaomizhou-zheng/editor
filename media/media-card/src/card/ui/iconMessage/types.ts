import { ReactNode } from 'react';
import { MessageDescriptor } from 'react-intl-next';

export type InternalIconMessageProps = {
  messageDescriptor: MessageDescriptor;
  animated?: boolean;
  reducedFont?: boolean;
};

export type CreatingPreviewProps = {
  disableAnimation?: boolean;
};

export type StyledTextProps = {
  animated?: boolean;
  reducedFont?: boolean;
};

export type IconMessageWrapperProps = {
  animated?: boolean;
  reducedFont?: boolean;
  children?: ReactNode;
};
