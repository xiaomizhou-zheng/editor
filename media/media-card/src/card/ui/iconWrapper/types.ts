import { Breakpoint } from '../common';

export type IconWrapperProps = {
  hasTitleBox: boolean;
  breakpoint: Breakpoint;
  children?: any;
};
