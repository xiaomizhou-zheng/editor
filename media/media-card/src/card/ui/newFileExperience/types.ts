import React, { MouseEvent, RefObject } from 'react';

import { UIAnalyticsEvent } from '@atlaskit/analytics-next';

import { CardDimensions, CardAppearance } from '../../../types';
import { Breakpoint } from '../common';
import { MediaCardCursor } from '../../../types';

export interface NewFileExperienceWrapperProps {
  testId?: string;
  breakpoint: Breakpoint;
  mediaCardCursor?: MediaCardCursor;
  dimensions?: CardDimensions;
  appearance?: CardAppearance;
  onClick?: (
    event: React.MouseEvent<HTMLDivElement>,
    analyticsEvent?: UIAnalyticsEvent,
  ) => void;
  onMouseEnter?: (event: MouseEvent<HTMLDivElement>) => void;
  mediaType?: string;
  disableOverlay: boolean;
  displayBackground: boolean;
  selected: boolean;
  isPlayButtonClickable: boolean;
  isTickBoxSelectable: boolean;
  shouldDisplayTooltip: boolean;
  innerRef?: RefObject<HTMLDivElement>;
  children?: JSX.Element;
}
