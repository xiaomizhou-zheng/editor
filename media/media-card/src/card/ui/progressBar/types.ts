import { Breakpoint } from '../common';

export type StyledBarProps = {
  progress?: number;
  breakpoint: Breakpoint;
  positionBottom: boolean;
  showOnTop: boolean;
};
