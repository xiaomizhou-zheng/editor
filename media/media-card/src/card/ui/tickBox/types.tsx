export type TickBoxProps = {
  selected?: boolean;
  children?: any;
};
