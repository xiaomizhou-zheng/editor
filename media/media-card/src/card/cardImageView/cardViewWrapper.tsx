/**@jsx jsx */
import { jsx } from '@emotion/react';
import { WrapperProps } from '../types';
import { wrapperStyles } from '../styles/styles';

export const Wrapper = (props: WrapperProps) => {
  const {
    testId,
    shouldUsePointerCursor,
    breakpointSize,
    dimensions,
    onClick,
    onMouseEnter,
    innerRef,
  } = props;
  return (
    <div
      id="cardViewWrapper"
      data-testid={testId}
      css={wrapperStyles({
        shouldUsePointerCursor,
        breakpointSize,
        dimensions,
      })}
      onClick={onClick}
      onMouseEnter={onMouseEnter}
      ref={innerRef}
    >
      {props.children}
    </div>
  );
};
