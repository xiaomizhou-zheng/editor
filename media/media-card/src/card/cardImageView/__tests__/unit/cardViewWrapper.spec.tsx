import React from 'react';
import { mount } from 'enzyme';

import { Wrapper } from '../../cardViewWrapper';
import { CardDimensions } from '../../../../types';

describe('Root Wrapper', () => {
  const dimensions: CardDimensions = {
    width: 400,
    height: 300,
  };
  const cardViewWrapperId = 'div#cardViewWrapper';

  describe('File', () => {
    it('should render properly with default properties', () => {
      const file = mount(<Wrapper />).find(cardViewWrapperId);

      expect(file).toMatchSnapshot();
    });

    it('should render properly with passed dimensions', () => {
      const defaultWithDimensions = mount(
        <Wrapper dimensions={dimensions} />,
      ).find(cardViewWrapperId);
      const auto = mount(<Wrapper dimensions={dimensions} />).find(
        cardViewWrapperId,
      );

      expect(defaultWithDimensions).toMatchSnapshot();
      expect(auto).toMatchSnapshot();
    });

    it('should apply breakpoint rules bassed on breakpointSize', () => {
      const small = mount(<Wrapper breakpointSize="small" />).find(
        cardViewWrapperId,
      );
      const medium = mount(<Wrapper breakpointSize="medium" />).find(
        cardViewWrapperId,
      );
      const large = mount(<Wrapper breakpointSize="large" />).find(
        cardViewWrapperId,
      );
      const xlarge = mount(<Wrapper breakpointSize="xlarge" />).find(
        cardViewWrapperId,
      );

      expect(small).toMatchSnapshot();
      expect(medium).toMatchSnapshot();
      expect(large).toMatchSnapshot();
      expect(xlarge).toMatchSnapshot();
    });

    it('should render right cursor when shouldUsePointerCursor is passed', () => {
      const withCursor = mount(<Wrapper shouldUsePointerCursor />).find(
        cardViewWrapperId,
      );
      const withoutCursor = mount(
        <Wrapper shouldUsePointerCursor={false} />,
      ).find(cardViewWrapperId);

      expect(withCursor).toMatchSnapshot();
      expect(withoutCursor).toMatchSnapshot();
    });
  });
});
