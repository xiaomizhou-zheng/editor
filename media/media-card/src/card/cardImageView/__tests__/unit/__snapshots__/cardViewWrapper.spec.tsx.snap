// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Wrapper File should apply breakpoint rules bassed on breakpointSize 1`] = `
.emotion-0 {
  cursor: default;
}

.emotion-0 .title {
  font-size: 12px;
}

.emotion-0 .file-type-icon span {
  width: 14px!important;
  height: 14px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should apply breakpoint rules bassed on breakpointSize 2`] = `
.emotion-0 {
  cursor: default;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should apply breakpoint rules bassed on breakpointSize 3`] = `
.emotion-0 {
  cursor: default;
}

.emotion-0 .overlay {
  padding: 24px;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-size {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 18px!important;
  height: 18px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should apply breakpoint rules bassed on breakpointSize 4`] = `
.emotion-0 {
  border-radius: 2px;
  cursor: default;
}

.emotion-0 .title {
  font-size: 16px;
}

.emotion-0 .file-size {
  font-size: 14px;
}

.emotion-0 .wrapper,
.emotion-0 .img-wrapper {
  border-radius: 2px;
}

.emotion-0 .overlay {
  padding: 24px;
}

.emotion-0 .file-type-icon span {
  width: 18px!important;
  height: 18px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should render properly with default properties 1`] = `
.emotion-0 {
  cursor: default;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should render properly with passed dimensions 1`] = `
.emotion-0 {
  height: 300px;
  max-height: 100%;
  width: 400px;
  max-width: 100%;
  cursor: default;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should render properly with passed dimensions 2`] = `
.emotion-0 {
  height: 300px;
  max-height: 100%;
  width: 400px;
  max-width: 100%;
  cursor: default;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should render right cursor when shouldUsePointerCursor is passed 1`] = `
.emotion-0 {
  cursor: pointer;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;

exports[`Root Wrapper File should render right cursor when shouldUsePointerCursor is passed 2`] = `
.emotion-0 {
  cursor: default;
}

.emotion-0 .title {
  font-size: 14px;
}

.emotion-0 .file-type-icon span {
  width: 16px!important;
  height: 16px!important;
}

<div
  className="emotion-0"
  id="cardViewWrapper"
/>
`;
